### DISCUSSION TITLE
Two devices with the same VendorID & ProductID

### DISCUSSION THREAD

**Discussion by mugnuff** [pyusb/pyusb#459] [Q&A] [answered]
Hello,

I'm trying to measure the inaccuracy of a step motor by using a laser to measure the distance it travelled.
I'm using PySerial for the laser, which is connected by USB with a RS422 to USB converter, which works just fine. The motor control is connected via USB aswell and I was planning to use PyUSB with libusb to tell the motor to go any amount of stept and return the value of the amount of steps to calculate the theoretical value of the distance. I tried to find the VendorID and the ProductID of the motor control by using this code:

```
import sys
import usb.core
dev = usb.core.find(find_all=True)
for cfg in dev:
sys.stdout.write('Decimal VendorID=' + str(cfg.idVendor) + ' & ProductID=' + str(cfg.idProduct) + '\n')
sys.stdout.write('Hexadecimal VendorID=' + hex(cfg.idVendor) + ' & ProductID=' + hex(cfg.idProduct) + '\n\n')
```
The output looks like this:
```
Decimal VendorID=1027 & ProductID=24577
Hexadecimal VendorID=0x403 & ProductID=0x6001
```
Besides all the other connected USB Devices I found that both, the motor control and the laser, use a FTDI chip and they both have the same VendorID and ProductID. So if Itry to print usb.core.find(idVendor=0x0403, idProduct=0x6001) it doesn't output the data from the motor control but the laser. I guess removing one of them is a possible solution but I need both of them connected at the same time. Is there a possibility (or an argument of usb.core.find) to use the bus and address or the serial number instead of the Vendor and ProductID?
In the PyUSB tutorial it said that it was possible but I still couldn't figure it out.
Thanks a lot!

**Comment by jonasmalacofilho** [comment, marked as answer]
You can filter by bus/address or serial number (or other similar properties) by supplying a `custom_match` function to `find(...)`, or by filtering the results of `find(..., find_all=True)` yourself.

For an example of how to use `custom_match`, see the documentation of `find(...)`:

```
$ pydoc usb.core.find
```

**Reply by mcuee** [reply]
The FTDI adapter will usually have serial number, so it is probably easier to use serial number to differentiate the two adpaters.
